src/udev/dmi_memory_id/dmi_memory_id.c

```diff
@@ -417,9 +417,10 @@ static void dmi_memory_device_size_detail(
                 dmi_print_memory_size("MEMORY_DEVICE", attr_suffix, slot_num, code, MEMORY_SIZE_UNIT_BYTES);
 }
 
+static unsigned next_slot_num = 0;
```

Comment:
hmm, actually, we don't like global variables really. It's fine for stuff that is inherently process-global (i.e. the parsed out fields from the command line or the environment block or so), but for other stuff we generally try to avoid it. AFAICS it's pretty easy to get rid of this one, just make this a local variable in dmi_table_decode() and pass a pointer to dmi_decode().

It's mostly a matter of coding style/hygiene, so that we can copy stuff around in our tree/reuse it without too much work.

I am aware this was a global variable before, too, just one hidden in a function. But if this is touched, let's fix this properly, please